### PR TITLE
Fixed issue where SSSD wasn't started after an auth configmap update.

### PR DESCRIPTION
**Pull request by abellotti**
- Attempt to enable and starting during initialize-httpd-auth service is too early in the process.
- Created an enable-sssd service which we run after both initialize-httpd-auth and httpd which enables and start sssd if the /etc/sssd/sssd.conf file was overlayed.
- Simplified the logic in initialize-httpd-auth since that was moved out.

### CODE REVIEW

**Comment by abellotti** [comment]
Needed this fix for the ldap (ext-auth w/sssd) tests in https://github.com/ManageIQ/manageiq-pods/pull/210

**Comment by carbonin** [comment]
Can we use a conditional in the sssd unit on sssd.conf instead of trying to delay the start up?

Or is it okay to just always enable it and let it fail if the file isn't present?

**Comment by abellotti** [comment]
I think so, I can add a ConditionPathExists on the sssd.service Unit and always enable it. As long as systemd does not do the unit condition checks until it is time to start the service, it should work.  I'll give that a try, Thanks !!.

**Comment by abellotti** [comment]
@carbonin all good, nothing a little After and ConditionPathExists couldn't help.

screen shot 2017-09-08 at 11 27 41 am

screen shot 2017-09-08 at 11 54 25 am

diffs: https://github.com/ManageIQ/container-httpd/pull/16/commits/8ffa545b4109cd3601a2f0612799d5bbf81ab7b6

File: Dockerfile [hunk 1]
```diff
@@ -69,6 +69,10 @@ COPY docker-assets/initialize-httpd-auth.sh   /usr/bin
 
 COPY docker-assets/initialize-httpd-auth.service /usr/lib/systemd/system/initialize-httpd-auth.service
 
+## Make sure sssd has the right startup conditions
+RUN  mkdir -p /etc/systemd/system/sssd.service.d
+COPY docker-assets/sssd-startup.conf      /etc/systemd/system/sssd.service.d/startup.conf
```
Comment (carbonin) [inline Dockerfile]: Why the extra spacing here?

**Review by carbonin — APPROVED** [review]
Looks good to me other than the one extra spacing.

File: Dockerfile (hunk 1 above)
Comment (abellotti) [reply Dockerfile]: sorry, was aligned with the lower COPY, will update.

**Pull request merged**